text/0198-slice-notation.md

```diff
@@ -0,0 +1,148 @@
+- Start Date: 2014-09-11
+- RFC PR #: [rust-lang/rfcs#198](https://github.com/rust-lang/rfcs/pull/198)
+- Rust Issue #: [rust-lang/rust#17177](https://github.com/rust-lang/rust/issues/17177)
+- Translators: [[@FizzyElt](https://github.com/FizzyElt)]
+- Commit: [The commit link this page based on](https://github.com/rust-lang/rfcs/blob/4009b546172c558a1cfa0f39dd81c896312f73d5/text/0198-slice-notation.md)
+- Updated: 2022-05-27
+
+# 概要
+
+此 RFC 增加了*多載切片符號*：
+
+- `foo[]` 對應 `foo.as_slice()`
+- `foo[n..m]` 對應 `foo.slice(n, m)`
+- `foo[n..]` 對應 `foo.slice_from(n)`
+- `foo[..m]` 對應 `foo.slice_to(m)`
+- 以上所有的 `mut` 變體
+
+透過兩個新特徵（trait），`Slice` 跟 `SliceMut`.
+
+它還將 範圍`配對`模式的符號改為 `...`，以表示它們是包含的，而 `..` 在切片中是不包含的。
+
+# 動機
+
+引入此功能有兩個主要動機。
+
+### 符合人體工學
+
+在切片操作，特別是 `as_slice`，是相當常見的基本操作，用來處理 vector，也可以用來處理其他類型的容器。我們已經有了透過 `Index` 特徵進行索引的符號，此 RFC 本質上是這特徵的延續。
+
+`as_slice` 操作符尤其重要，自從我們已經擺脫以強迫的方式自動切片，顯式的 `as_slice` 調用變得非常普遍，這也是語言中[人體工學優先/第一印象](https://github.com/rust-lang/rust/issues/14983)的問題之一。還有一些其他方法可以解決這個特定問題，但這些替代方法具有下方討論的一些缺點（參閱"替代方案"）。
```

Comment:
如果選人體工學，我可能更偏好「人因工程」這種冷僻字~~讓它更混亂~~，人體工學聽起來太物理了，人因工程才有以人為出發點的感覺，比較文雅。

Ref: [韋氏辭典 ergonomics](https://www.merriam-webster.com/dictionary/ergonomics)

> _an applied science concerned with designing and arranging things people use so that the people and things interact most efficiently and safely_